editor/src/core/model/element-metadata-utils.ts

```diff
@@ -2321,44 +2322,80 @@ function fillGlobalContentBoxFromAncestors(
   return workingElements
 }
 
-// When a conditional has no siblings, and there is a js expression in its active branch, its globalFrame/localFrame
-// should be inherited from its parent
+// There is a case when conditionals should inherit their globalFrame from their parent:
+// - when a conditional has no siblings
+// - when the active branch is a js expression
+// - when that expression has no elementsWithin (so it is a leaf element in the hierarchy)
+// We also allow embedded conditionals where the deepest one has the correct active branch and
+// none of the conditionals have siblings.
+// Why is this useful? Because these expressions don't have globalFrame, so their conditional parent do not have
+// a globalFrame neither. But when these conditionals only contain text expressions, and they don't have siblings,
+// it make sense to treat the whole parent element of the conditional as a container for the text expression itself.
+// Example: In the case `<div>Hello</div>`, we treat the whole div as text, we do not differentiate between the size of
+// Hello and the size of the div.
+// When we have the element `<div>{true ? 'Hello' : <div />}</div>`, the conditional appears between the <div> parent and
+// the Hello text, but we still want to be able to select and text edit the text content.
+// Maybe in the future we will treat text expressions as elefants, and they will have a measured size, but until then this is
+// a good compromise to give these conditionals/expressions selectability and editability on the canvas.
 function fillConditionalGlobalFrameFromAncestors(
   metadata: ElementInstanceMetadataMap,
 ): ElementInstanceMetadataMap {
   const workingElements = { ...metadata }
 
   const conditionalsWithNoSiblingsAndExpressionInActiveBranch = Object.keys(workingElements).filter(
     (p) => {
-      const element = workingElements[p]
-      if (element.conditionValue === 'not-a-conditional') {
-        return false
-      }
-      const condElement =
-        isRight(element.element) && isJSXConditionalExpression(element.element.value)
-          ? element.element.value
-          : null
+      const isConditionalsWithNoSiblingsAndExpressionInActiveBranch = (
```

Comment:
can I suggest to rename this to avoid the potential misunderstanding `InActive === Inactive`?

Reply:
oh, I did not think about this 🤦